jms/deployment/src/main/java/io/quarkus/artemis/jms/deployment/DevServicesArtemisJmsProcessor.java

```diff
@@ -0,0 +1,231 @@
+import io.quarkus.deployment.logging.LoggingSetupBuildItem;
+import io.quarkus.devservices.common.ConfigureUtil;
+import io.quarkus.devservices.common.ContainerAddress;
+import io.quarkus.devservices.common.ContainerLocator;
+import io.quarkus.runtime.configuration.ConfigUtils;
+
+/**
+ * Start a ActiveMQ Artemis broker if needed
+ */
+public class DevServicesArtemisJmsProcessor {
+    private static final Logger LOGGER = Logger.getLogger(DevServicesArtemisJmsProcessor.class);
+    private static final String QUARKUS_ARTEMIS_URL = "quarkus.artemis.url";
+
+    /**
+     * Label to add to shared Dev Service for ActiveMQ Artemis running in containers.
+     * This allows other applications to discover the running service and use it instead of starting a new instance.
+     */
+    static final String DEV_SERVICE_LABEL = "quarkus-dev-service-artemis";
+    static final int ARTEMIS_PORT = 61616;
+
+    private static final ContainerLocator artemisContainerLocator = new ContainerLocator(DEV_SERVICE_LABEL, ARTEMIS_PORT);
+
+    static volatile RunningDevService devService;
+    static volatile ArtemisDevServiceCfg cfg;
+    static volatile boolean first = true;
+
+    @BuildStep(onlyIfNot = IsNormal.class, onlyIf = GlobalDevServicesConfig.Enabled.class)
+    public DevServicesResultBuildItem startArtemisDevService(
+            DockerStatusBuildItem dockerStatusBuildItem,
+            LaunchModeBuildItem launchMode,
+            ArtemisBuildTimeConfig artemisBuildTimeConfig,
+            List<DevServicesSharedNetworkBuildItem> devServicesSharedNetworkBuildItem,
+            Optional<ConsoleInstalledBuildItem> consoleInstalledBuildItem,
+            CuratedApplicationShutdownBuildItem closeBuildItem,
+            LoggingSetupBuildItem loggingSetupBuildItem, GlobalDevServicesConfig devServicesConfig) {
+
+        ArtemisDevServiceCfg configuration = getConfiguration(artemisBuildTimeConfig);
+
+        if (devService != null) {
+            boolean shouldShutdownTheBroker = !configuration.equals(cfg);
+            if (!shouldShutdownTheBroker) {
+                return devService.toBuildItem();
+            }
+            shutdownBroker();
+            cfg = null;
+        }
+
+        StartupLogCompressor compressor = new StartupLogCompressor(
+                (launchMode.isTest() ? "(test) " : "") + "ActiveMQ Artemis Dev Services Starting:",
+                consoleInstalledBuildItem, loggingSetupBuildItem);
+        try {
+            devService = startArtemis(dockerStatusBuildItem, configuration, launchMode,
+                    !devServicesSharedNetworkBuildItem.isEmpty(),
+                    devServicesConfig.timeout);
+            if (devService == null) {
+                compressor.closeAndDumpCaptured();
+            } else {
+                compressor.close();
+            }
+        } catch (Throwable t) {
+            compressor.closeAndDumpCaptured();
+            throw new RuntimeException(t);
```

Comment:
```java
throw (t instanceof RuntimeException ? (RuntimeException) t : t
```

Reply:
I think it should be wrapped with `RuntimeException`.

Reply:
Ah yes, I missed that. Should be:

```java
throw (t instanceof RuntimeException ? (RuntimeException) t : new RuntimeException(t);
```

the idea is to not unnecessarily wrap exceptions in `RuntimeException`.